api/apps/geoprocessing/src/import/pieces-importers/project-metadata.piece-importer.ts

```diff
@@ -19,13 +19,75 @@ export class ProjectMetadataPieceImporter implements ImportPieceProcessor {
     private readonly fileRepository: FileRepository,
     @InjectEntityManager(geoprocessingConnections.apiDB)
     private readonly entityManager: EntityManager,
-  ) {}
+    private readonly logger: Logger,
+  ) {
+    this.logger.setContext(ProjectMetadataPieceImporter.name);
+  }
 
   isSupported(piece: ClonePiece): boolean {
     return piece === ClonePiece.ProjectMetadata;
   }
 
+  private async getRandomOrganizationId(): Promise<string> {
+    const [{ id }]: [{ id: string }] = await this.entityManager.query(`
+      SELECT id FROM organizations LIMIT 1
+    `);
+    return id;
+  }
+
   async run(input: ImportJobInput): Promise<ImportJobOutput> {
-    throw new Error('Missing implementation');
+    const { uris, resourceId } = input;
+
+    if (uris.length !== 1) {
+      const errorMessage = `uris array has an unexpected amount of elements: ${uris.length}`;
+      this.logger.error(errorMessage);
+      throw new Error(errorMessage);
+    }
+    const [projectMetadataLocation] = uris;
+
+    const readableOrError = await this.fileRepository.get(
+      projectMetadataLocation.uri,
+    );
+    if (isLeft(readableOrError)) {
+      const errorMessage = `Zip file not found: ${projectMetadataLocation.uri}`;
+      this.logger.error(errorMessage);
+      throw new Error(errorMessage);
+    }
+
+    const stringProjectMetadataOrError = await extractFile(
+      readableOrError.right,
+      projectMetadataLocation.relativePath,
+    );
+    if (isLeft(stringProjectMetadataOrError)) {
+      const errorMessage = `Project metadata file extraction failed: ${projectMetadataLocation.relativePath}`;
+      this.logger.error(errorMessage);
+      throw new Error(errorMessage);
+    }
+
+    // TODO OrganizationId should be specified when creating the Import
```

Comment:
```suggestion
    // TODO As we don't handle organizations for the time being,
    // the imported/cloned project is homed arbitrarily within an
    // existing organization. Once proper handling of organizations
    // is added, users may be able to specify within which organization
    // an imported/cloned project should be created.
```

Reply:
just suggesting to add some detail to this comment as this will puzzle whoever reads this a few months from now, unless they ask you or me 😅